app/Models/User.php

```diff
@@ -58,4 +63,8 @@ class User extends Authenticatable
     protected $hidden = [
         'password',
     ];
+
+    public function posts(): HasMany {
+        return $this->hasMany(Post::class, 'author_id');
```

Comment:
Ah yes, there are different names for the foreign keys 😱. Maybe previously no error was found because the query did not use "Eloquent relationships" but used join.